### DISCUSSION TITLE
Find max task available

### DISCUSSION THREAD

**Discussion by vuphamngoctan1612** [google/or-tools#4295] [CP-SAT questions]
Similar assignment problem, i have a list of worker, 1 have skills
I have 1 type service, this type service require any skills
Worker have to have enough skills to map with a service
I want to find max service with list of worker. What should i do?

**Comment by sschnug** [comment]
- Work through [Wikipedias Assignment-Problem Article (focus on LP model)](https://en.wikipedia.org/wiki/Assignment_problem#:~:text=1%5D%3A%E2%80%8A6-,Solution%20by%20linear%20programming,-%5Bedit%5D) to get a grasp about the basic modelling through linear-constraints
  - Core concepts: linear-constraints over (0,1) (or there [0,1]) variables for:
    - *exact covering / hitting*: `=1`
    - *covering*  -> `>= 1`
    - *packing* -> `<=1`
- Work through the [examples available within the ortools project](https://github.com/google/or-tools/tree/stable/examples) 
  - Maybe [assignment_with_constraints_sat](https://github.com/google/or-tools/blob/stable/examples/python/assignment_with_constraints_sat.py) (but i did not analyze them against your description)
- Assume, that **you really want to base your model on a matrix of boolean variables!**
- Consider preprocessing skill-disqualification:
  - Worker w does not cover skills for task t0 -> set matrix(w, t0) = 0 | aka forbid assignment
  - Same for w not covering skill for task t7 -> set matrix(w, t7) = 0 

This should be a good start as this is one of the more easy problems to formulate! 
You probably should not expect any copy-pastable code here as your question is lacking a lot.
